tools/wptrunner/wptrunner/browsers/webkitgtk_minibrowser.py

```diff
@@ -0,0 +1,74 @@
+from .base import get_timeout_multiplier   # noqa: F401
+from .webkit import WebKitBrowser
+from ..executors import executor_kwargs as base_executor_kwargs
+from ..executors.executorwebdriver import (WebDriverTestharnessExecutor,  # noqa: F401
+                                           WebDriverRefTestExecutor)  # noqa: F401
+from ..executors.executorwebkit import WebKitDriverWdspecExecutor  # noqa: F401
+
+__wptrunner__ = {"product": "webkitgtk_minibrowser",
+                 "check_args": "check_args",
+                 "browser": "WebKitGTKMiniBrowser",
+                 "browser_kwargs": "browser_kwargs",
+                 "executor": {"testharness": "WebDriverTestharnessExecutor",
+                              "reftest": "WebDriverRefTestExecutor",
+                              "wdspec": "WebKitDriverWdspecExecutor"},
+                 "executor_kwargs": "executor_kwargs",
+                 "env_extras": "env_extras",
+                 "env_options": "env_options",
+                 "run_info_extras": "run_info_extras",
+                 "timeout_multiplier": "get_timeout_multiplier"}
+
+
+def check_args(**kwargs):
+    pass
+
+
+def browser_kwargs(test_type, run_info_data, config, **kwargs):
+    return {"binary": kwargs["binary"],
+            "webdriver_binary": kwargs["webdriver_binary"],
+            "webdriver_args": kwargs.get("webdriver_args")}
+
+
+def capabilities(server_config, **kwargs):
+    args = kwargs.get("binary_args", [])
+    if "--automation" not in args:
+        args.append("--automation")
+
+    return {
+        "browserName": "MiniBrowser",
+        "browserVersion": "2.20",
```

Comment:
Is this the payload for the POST /session request? If so is the version required, given that the pass the binary it seems like nothing useful could be done with it?

Reply:
Not sure if its required. I copied it from the current "webkit" version. i will check if its needed.

Reply:
It seems `browserVersion` acts as the minimum version that would be allowed. So with this code, the webdriver would refuse to start the MiniBrowser if it reports to be a version lower than < 2.20 runninganything newer than "2.20". The f[irst release of webkitgtk to support webdriver was 2.18](https://blogs.igalia.com/carlosgc/2017/09/09/webdriver-support-in-webkitgtk-2-18/). So I guess it makes sense to lower this to 2.18

Reply:
Isn't it allowed to omit it entirely? I known that safaridriver at least is able to start a session with an empty capabilities dictionary. Not sure what the spec requires.

Reply:
Yes, it is allowed to omit the version entirely. If its omitted, then it will start the browser without checking the version.

Reply:
That sounds better, right?

Reply:
Right :)